src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -41,6 +44,21 @@ import {
     SettingsModalStackNavigator,
 } from './ModalStackNavigators';
 
+Onyx.connect({
+    key: ONYXKEYS.MY_PERSONAL_DETAILS,
+    callback: (val) => {
+        const timezone = lodashGet(val, 'timezone', {});
+        const currentTimezone = moment.tz.guess(true);
+
+        // If the current timezone is different than the user's timezone, and their timezone is set to automatic
+        // then update their timezone.
+        if (_.isObject(timezone) && timezone.automatic && timezone.selected !== currentTimezone) {
+            timezone.selected = currentTimezone;
+            PersonalDetails.setPersonalDetails({timezone});
+        }
+    },
+});
+
```

Comment:
Why is this block defined here? 

The code here would run as soon as this file is **imported** and it's not related to the actual `AtuhScreens` component lifecycle

It could have been added to `PersonalDetails` along with the existing static `Onix.connect` calls there?

https://github.com/Expensify/Expensify.cash/blob/9b0938441611f90ed8611513954258227dabbd35/src/libs/actions/PersonalDetails.js#L13-L23

I noticed this while resolving a merge conflict in `AtuhScreens` in #2346

Reply:
There is a larger conversation [here](https://github.com/Expensify/Expensify.cash/pull/2318#discussion_r610812435) about the discussion to move this logic to AuthScreens.js.

Reply:
The linked conversations is about setting the timezone should not be a side effect of the `format` function and should be moved out of it.

Ok it was moved out but you can literally copy this whole block, put it anywhere and it will still work

https://github.com/Expensify/Expensify.cash/blob/87c3b3dbf5045b196eacf2d89554a3cd2d74586f/src/libs/Navigation/AppNavigator/AuthScreens.js#L47-L60

I just don't get why it's in `AuthScreens` it's not related to anything in this file. 

That's why I suggested that it can be housed in `PersonalDetails` below the other `Onyx.connect` calls there

Reply:
Oh, that's a good point. It indeed can be anywhere. I am fine with putting it in the PersonalDetails file

Reply:
Ah, I see. Yeah, it could be in PersonalDetails.js. @kidroca if you feel strongly about it you can modify it in your PR (or a future PR), or I will modify it in my next E.cash change.

Reply:
I could, but then git would annotate me as the author of the code, while I don't know how it actually works. It would be best if the author (or at least someone involved in this) moves it